.github/workflows/format.yml

```diff
@@ -0,0 +1,47 @@
+name: Format
+on:
+  issue_comment:
+    types: [created]
+jobs:
+  edit:
+    name: clang-format
+    if: ${{ github.event.comment.body == '@crtrott format' }}
+    runs-on: ubuntu-latest
+    env:
+      CLANG_FORMAT: clang-format-8
+    steps:
+      - name: Check if message comes from a PR
+        uses: octokit/request-action@v2.x
+        id: issue_info
+        with:
+          route: GET /repos/:repository/issues/:pull_number
+          repository: ${{ github.repository }}
+          pull_number: ${{ github.event.issue.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get PR info
+        if: ${{ fromJSON(steps.issue_info.outputs.data).pull_request }}
+        uses: octokit/request-action@v2.x
+        id: pr_info
+        with:
+          route: GET /repos/:repository/pulls/:pull_number
+          repository: ${{ github.repository }}
+          pull_number: ${{ github.event.issue.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout head branch from PR
+        if: ${{ fromJSON(steps.issue_info.outputs.data).pull_request }}
+        uses: actions/checkout@master
+        with:
+          repository: ${{ fromJSON(steps.pr_info.outputs.data).head.repo.full_name }}
+          ref: ${{ fromJSON(steps.pr_info.outputs.data).head.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run clang-format
+        if: ${{ fromJSON(steps.issue_info.outputs.data).pull_request }}
+        run: find . -type f -regex ".*\.\(c\|h\|cc\|cpp\|hpp\)" -print0 | xargs -0 ${CLANG_FORMAT} -i -style=file
+      - name: Commit and push
+        if: ${{ fromJSON(steps.issue_info.outputs.data).pull_request }}
+        run: |
+          git config --global user.name "Christian Trott"
+          git config --global user.email "crtrott@sandia.gov"
```

Comment:
I think this ought to be a bot that commit or possibly the person that commented on the pull request.

Reply:
One can get the username out of `github.event.comment.user.login`, but then one would need to lookup the user email/ full name, which is already public.

Reply:
> The latter approach (having a command for formatting all the commits individually) seems more reasonable to me.

That won't work in most cases, running `clang-format` on every commit will most likely lead to conflicts during the rebase, so I don't think this can be done automatically easily.

Reply:
https://stackoverflow.com/a/45115656